examples/LLM_Workflows/knowledge_retrieval/state.py

```diff
@@ -137,7 +137,7 @@ def call_arxiv_function(messages, full_message):
             return response
         except Exception as e:
             logger.error(type(e))
-            raise Exception("Function chat request failed")
+            raise Exception("Function chat request failed") from e
```

Comment:
1. Do we not know what sort of exception is raised by the previous code?
2. Can we not raise a [more specific exception](https://docs.python.org/3/library/exceptions.html#exception-hierarchy)? Even if it's a `RuntimeError`...?

Reply:
Agreed. But this is an example -- so low stakes.